src/ops/array.rs

```diff
@@ -15,42 +15,46 @@ pub(crate) fn rescale_internal(value: &mut [u32; 3], value_scale: &mut u32, new_
     }
 
     if *value_scale > new_scale {
-        let mut diff = *value_scale - new_scale;
+        let mut diff = value_scale.wrapping_sub(new_scale);
```

Comment:
`wrapping_sub` is explicitly telling the intended arithmetic operation behavior, i.e., it is self documented. This will help external users to have a better glimpse of what is going on